website/pages/docs/connect/observability/ui-visualization.mdx

```diff
@@ -0,0 +1,501 @@
+URLs](#configuring-dashboard-urls).
+
+It is possible to configure the UI to fetch basic metrics from your metrics
+provider storage to augment the visualization as displayed below.
+
+TODO: realistic screenshot here
+
+Consul has built-in support for overlaying metrics from a
+[Prometheus](https://prometheus.io) backend. Alternative metrics providers may
+be supported using a new and experimental Javascript API. See [Custom Metrics
+Providers](#custom-metrics-providers).
+
+
+## Configuring the UI To Display Metrics
+
+To configure Consul's UI to be able to fetch metrics there are several
+configuration variables needed. These need to be set on each Consul Agent that
+is responsible for serving the UI. If there are multiple clients with the UI
+enabled in a Datacenter for redundancy, these configurations should be added to
+all of them.
+
+We assume that the UI is already enabled which since Consul 1.9.0 is done by
+setting [`ui_config.enabled`](/docs/agent/options#ui_config_enabled) to `true`
+in the agent's configuration file.
+
+To use the built-in Prometheus provider,
+[`ui_config.metrics_provider`](/docs/agent/options#ui_config_metrics_provider)
+must be set to `prometheus`.
+
+Finally, the UI currently must query the metrics provider through a proxy
+endpoint to simplify setup where Prometheus isn't already publicly available to
+user's browsers directly.
+
+To set this up, provide the URL that the _Consul agent_ can reach the Prometheus
+server on in
+[`ui_config.metrics_proxy.base_url`](/docs/agent/options#ui_config_metrics_proxy_base_url).
+For example in Kubernetes, the Prometheus helm chart by default installs a
+service named `prometheus-server` so each Consul agent can reach it on
+`http://prometheus-server` (using Kubernetes' DNS resolution).
+
+A full configuration to enable Prometheus is given below.
+
+```hcl
+ui_config {
+  enabled = true
+  metrics_provider = "prometheus"
+  metrics_proxy {
+    base_url = "http://prometheus-server"
+  }
+}
+```
+
+## Configuring Dashboard URLs
+
+Since Consul's visualization is intended as an overview of your mesh and not a
+comprehensive monitoring tool, you can configure a service dashboard URL
+template which allows users to click directly through to the relevant
+service-specific dashboard in an external tool like Grafana or a hosted
+provider.
+
+To configure this, you must provide a URL template in the [agent configuration
+file](/docs/agent/options#ui_config_dashboard_url_templates) for all agents that
+have the UI enabled. The template is essentially the URL to the external
+dashboard, but can have placeholder values which will be replaced with the
+service name, namespace and datacenter where appropriate to allow deep-linking
+to the relevant information.
+
+An example with Grafana is shown below.
+
+```hcl
+ui_config {
+  enabled = true
+  dashboard_url_templates {
+    service = "https://grafana.example.com/d/lDlaj-NGz/ ⏎
```

Comment:
I don't think these `⏎` characters was intended?

Reply:
They were to show that I was manually word-wrapping the long line. Does that seem confusing? The `pre` tag will allow the line to continue off the page otherwise which makes it hard to see the actual interpolation in the URL, but since it's one string in HCL I didn't want to break it into a multiline string which would also break.

It does mean it's not copy-pastable but I feel the value of this is showing the example rather than being something to copy.

Woudl be be clearer to add a note after like:

 > Note ⏎ indicates word-wrapping added to make the long URL line easier to read but it must be a single value in HCL

Or something?

Reply:
I removed them and made a note not to copy the line breaks:

![image](https://user-images.githubusercontent.com/120915/98398943-a35ea880-2059-11eb-8b96-47df2c5d0a3a.png)